tests/Functional/src/com/microsoft/aad/adal/test/AuthenticationContextTest.java

```diff
@@ -1364,6 +1367,98 @@ public void testRefreshTokenRequestWithFamilyIdFailed() throws InterruptedExcept
+         addTokenCacheItemInCache(tokenItem2, mockCache);
+         
+         // token cache item 3 with FoCI flag
+         final TokenCacheItem tokenItem3 = createTokenCacheItemWithFoCIFlag("authority3", "clientid3", "accessToken3", "refreshToken3", 
+                 TEST_IDTOKEN_USERID, TEST_IDTOKEN_UPN);
+         addTokenCacheItemInCache(tokenItem3, mockCache);
+         
+         // also add a reglar token cache item with no FoCI flag, but with timestamp
+         final TokenCacheItem tokenItem4 = createTokenCacheItemWithFoCIFlag("authority4", "clientid4", "accessToken4", "refreshToken4", 
+                 TEST_IDTOKEN_USERID, TEST_IDTOKEN_UPN);
+         tokenItem4.setFamilyClientId(null);
+         final Calendar tokenUpdatedTime4 = new GregorianCalendar();
+         tokenUpdatedTime4.add(Calendar.MONTH, 3);
+         addTokenCacheItemInCache(tokenItem4, mockCache);
+         
+         final AuthenticationContext context = getAuthenticationContext(mockContext,
+                 VALID_AUTHORITY, false, mockCache);
+         
+         setConnectionAvailable(context, true);
+         MockWebRequestHandler webrequest = new MockWebRequestHandler();
+         
+         // Do silent token request and return idtoken in the result
+         final String responseBody = "{\"id_token\":\""
+                 + TEST_IDTOKEN
+                 + "\",\"access_token\":\"I am a new access token\",\"token_type\":\"Bearer\",\"expires_in\":\"10\",\"expires_on\":\"1368768616\",\"refresh_token\":\"I am a new refresh token\",\"scope\":\"*\"}";
+         webrequest.setReturnResponse(new HttpWebResponse(400, responseBody.getBytes(), null));
+         ReflectionUtils.setFieldValue(context, "mWebRequest", webrequest);
+         
+         final String anotherClientId = "anotherClientId";
+         try {
+             context.acquireTokenSilentSync("resource", anotherClientId,
+                     TEST_IDTOKEN_USERID);
+         } catch (AuthenticationException authException) {
+             assertEquals("Error code is not as expected", ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, authException.getCode());
+         }
+         
+         // Token cache item 2 is the one with latest timestamp, make sure refreshToken2 is used for making the request 
+         final String silentRequestPostMessage = webrequest.getRequestContent();
+         assertTrue(silentRequestPostMessage.contains("refresh_token=refreshToken2"));
+         assertFalse(silentRequestPostMessage.contains("refresh_token=refreshToken1"));
+         assertFalse(silentRequestPostMessage.contains("refresh_token=refreshToken3"));
+         assertFalse(silentRequestPostMessage.contains("refresh_token=refreshToken4"));
+         
+         clearCache(context);
+     }
+    
+    private TokenCacheItem createTokenCacheItemWithFoCIFlag(final String authority, final String clientId, final String accessToken, 
+        final String refreshToken, final String userId, final String displayableId) {
```

Comment:
Just create TokenCacheItemMock derived from TokenCacheItem  and override constructor